### ISSUE TITLE
Problem in some leap years...

### ISSUE THREAD

**Issue by ghost** [behrang/Ext.ux.JalaliDatePlugin#8]
Test case : 
1- input or select "1395/01/02"
2- after leaving the inputbox the date will be changed to "1395/01/01" (decreate one day)
another test case is "1389/01/02" .

As I debugged, the problem is with "**safeParse**" function inside "Ext.ux.JalaliDatePlugin" class where you write this:
` result = Ext.Date.add(parsedDate, Ext.Date.HOUR, -12);`

**Comment by behrang** [comment]
This is a really old project and I'm very busy now. Can you help fix this problem?

If I remember correctly, in the line you mentioned where 12 hours is subtracted, that's for timezone problems. Because in the first day of Jalaali year, daylight saving time hour is added, that day is buggy in conversions without time being specified. So 12 hours is added/subtracted to protect the conversion from daylight saving hour.

On a side note, are you using Asia/Tehran (+03:30) timezone?

**Comment by ghost** [comment]
Yes it is Tehran. Note that the problem is in leap years only.   I havnt tried to find any workaround yet. Maybe switching to UTC solves the problem but needed more investigation. have any clue?

Sent from my HTC

**Comment by behrang** [comment]
Can you first check the conversion algorithm? If that is correct, it might be a timezone problem. If algorithm isn't correct, then you may be want to migrate it to the new better conversion algorithm on [github.com/jalaali](https://github.com/jalaali).
